en/contributing/test.md

```diff
@@ -77,3 +87,138 @@ To run all telemetry tests:
 ```
 make && build/default/integration_tests_runner --gtest_filter="SitlTest.Telemetry*"
 ```
+
+## Writing Tests {#writing_tests}
+
+The [Google Test Primer](https://github.com/google/googletest/blob/master/googletest/docs/Primer.md)
+provides an excellent overview of how tests are written and used.
+
+Most of the existing plugins do not have unit tests, because we do not yet have the ability to [mock MAVLink communications](https://github.com/dronecore/DroneCore/issues/148) (needed to test most plugins). In consequence integration tests are used instead, 
+with the simulator providing appropriate MAVLink messages.
+
+> **Note** This section explains how to add tests for the default DroneCore code. 
+> Most of the information is also relevant to SDK/Extension plugin tests. See
+> [SDK Creation > Testing](../guide/sdk.md#testing) for information about how
+> to add tests for extensions to the test runner(s).
+
+
+### Writing Unit Tests
+
+Unit tests are stored as separate files in the same directory as their associated source code. Often they test the implementation (rather than the public API), and hence are named with the suffix **_impl_test.cpp**.
+
+
+#### Unit Test Files/Code
+
+Unit tests typically include the file to be tested, **dronecore.h**, and **gtest.h**. There are no standard shared test unit resources so 
+test functions are declared using `TEST`. All tests in a file should share the same test-case name (the first argument to `TEST`).
+
+The skeleton [example plugin unit test](https://github.com/dronecore/DroneCore/blob/master/external_example/plugins/example/example_impl_test.cpp) is shown below: 
+```cpp
+#include "example_impl.h"
+#include "dronecore.h"
+#include "global_include.h"
+#include <gtest/gtest.h>
+
+namespace dronecore {
+
+TEST(ExampleImpl, NoTest)
+{
+    ASSERT_TRUE(true);
+}
+
+} // namespace dronecore
+```
+
+#### Adding Unit Tests
+
+In order to run a unit test its file needs to be added to the DroneCore unit test program (`unit_tests_runner`). For the core plugins this is done in the main DroneCore [DroneCore/CMakeLists.txt](https://github.com/dronecore/DroneCore/blob/master/CMakeLists.txt#L187) file:
```

Comment:
@julianoes I think this is wrong - I was confused by the fact that the only unit tests we have are in the core. I think that for a plugin they should just define their unit tests in unit_tests_src in the same way as sdk plugins. Of course that doesn't explain what  `${plugin_unittest_source_files}` is for?